techniques/html/H-creating-dialogs-with-the-html-dialog-element.html

```diff
@@ -0,0 +1,212 @@
+      <ul>
+        <li>focus management into the newly-opened modal <code class="language-html">dialog</code>;</li>
+        <li>focus management back onto the triggering element (assuming the trigger is still on the page) when the modal <code class="language-html">dialog</code> is closed;</li>
+        <li>focus is constrained to the contents of the modal <code class="language-html">dialog</code> and the browser's chrome;</li>
+        <li>with no extra code, the underlying page is hidden from assistive technology when the modal <code class="language-html">dialog</code> is open;</li>
+        <li>the use of the <kbd>Escape</kbd> key to close the modal <code class="language-html">dialog</code>.</li>
+      </ul>
+      <p>This technique uses the HTML <code class="language-html">dialog</code> element rather than the ARIA equivalent.</p>
+   </section><section id="examples"><h2>Examples</h2>
+      <section class="example">
+         <h3>A dialog to sign up to a mailing list</h3>
+				 <p>This is an example of using a modal <code class="language-html">dialog</code> element to show a mailing-list sign-up form to a user. The main part of the page contains a <code class="language-html">button</code> element that, when activated, triggers the appearance of the modal <code class="language-html">dialog</code>. The <code class="language-html">button</code> uses the <code class="language-html">type</code> attribute to tell the browser that this isn't a <code class="language-html">submit</code> <code class="language-html">button</code>.</p>
+				 
+				 <p>When the modal <code class="language-html">dialog</code> is opened, the browser will hide all of the content on the underlying page from assistive technology. For example: a screen reader will not announce any of the content. Additionally, focusing with the keyboard is constrained to the <code class="language-html">dialog</code> element's focusable content and the browser's controls. The browser takes care of focus management for us by putting the focus onto the first focusable element in the dialog's <abbr title="Document Object Model">DOM</abbr>, which in this demo is the <code class="language-html">h1</code> element because it has a <code class="language-html">tabindex="-1"</code> attribute, making it focusable by scripts. Note that, although the <code class="language-html">dialog</code>'s Close <code class="language-html">button</code> is visibly before the <code class="language-html">h1</code>, it is the last item in the <code class="language-html">dialog</code>'s DOM. If the <code class="language-html">button</code> was first, it would receive focus when dialog was opened, which could result in the <code class="language-html">dialog</code> being closed by mistake if the <code class="language-html">button</code> was accidentally activated.</p>
```

Comment:
```suggestion
				 <p>When the modal <code class="language-html">dialog</code> is opened, the browser will hide all of the content on the underlying page from assistive technology. For example, a screen reader will not announce any of the content. Additionally, focusing with the keyboard is constrained to the <code class="language-html">dialog</code> element's focusable content and the browser's controls. The browser takes care of focus management for us by putting the focus onto the first focusable element in the dialog's <abbr title="Document Object Model">DOM</abbr>, which in this demo is the <code class="language-html">h1</code> element because it has a <code class="language-html">tabindex="-1"</code> attribute, making it focusable by scripts. Note that, although the <code class="language-html">dialog</code>'s Close <code class="language-html">button</code> is visibly before the <code class="language-html">h1</code>, it is the last item in the <code class="language-html">dialog</code>'s DOM. If the <code class="language-html">button</code> was first, it would receive focus when dialog was opened, which could result in the <code class="language-html">dialog</code> being closed by mistake if the <code class="language-html">button</code> was accidentally activated.</p>
```